crates/core/src/v2/mod.rs

```diff
@@ -0,0 +1,117 @@
+mod hrana;
+mod rows;
+mod statement;
+
+use std::sync::Arc;
+
+use crate::{Params, Result};
+pub use hrana::{Client, HranaError};
+
+pub use rows::{Row, Rows};
+use statement::LibsqlStmt;
+pub use statement::Statement;
+
+// TODO(lucio): Improve construction via
+//      1) Move open errors into open fn rather than connect
+//      2) Support replication setup
+enum DbType {
+    Memory,
+    Path { path: String },
+    Http { url: String },
+}
+
+pub struct Database {
+    db_type: DbType,
+}
+
+impl Database {
+    pub fn open_in_memory() -> Result<Self> {
+        Ok(Database {
+            db_type: DbType::Memory,
+        })
+    }
+
+    pub fn open(db_path: impl Into<String>) -> Result<Database> {
+        Ok(Database {
+            db_type: DbType::Path {
+                path: db_path.into(),
+            },
+        })
+    }
+
+    pub fn open_http(url: impl Into<String>) -> Result<Self> {
```

Comment:
Probably should be `open_remote()` because you pass `libsql://` URLs to it. Can be fixed later too.